### ISSUE TITLE
Duplicated Fabaceae names

### ISSUE THREAD

**Issue by mdoering** [CatalogueOfLife/data#602]
https://www.checklistbank.org/dataset/278910/names?q=Acacia%20caven
... and likely many more. Looks like WCVP *AND* WWW was included!

**Comment by yroskov** [comment]
@mdoering, indeed. Can you do something to prevent this for GSDs with unstable IDs?

>WCVP AND WWW

At least, there are no WWW vs WCVP-Fabaceae duplicates in the preview CoL-WCVP 2023-12-06.

**Comment by mdoering** [comment]
We could add automatic rematching whenever a dataset is imported. It would be a setting for a dataset just like we have for decisions. 

Currently the setting is on the dataset, that means if turned on it would rematch all sectors and all decisions from all projects. I think it would be better to have that as a setting on the project side. One global setting in the project to turn this feature on for every source. And one on the sector to override the global one, i.e. you can leave this setting off in the project but turn it on for selected sectors only. Would that help?

**Comment by mdoering** [comment]
The real fix is this one: https://github.com/CatalogueOfLife/backend/issues/1189
I will think and see how much time that would take, it is definitely a superior solution.

**Comment by yroskov** [comment]
>We could add automatic rematching 

I think, it will help. @gdower, what do you think?

**Comment by yroskov** [comment]
Ah, correction to my previous sentence: I think, it will help with sectors. But will it help with broken decisions? 

(In the case of WCVP-Fabaceae with unstable IDs, 8 their genera are blocked as taxa. These 8 genera are nested from WWW).

**Comment by mdoering** [comment]
I think it would, yes. There might still be some broken ones left in case of true removal of records or ambiguous duplicates. So the always need to be checked before a sync. But this should solve the bulk of problems.

**Comment by mdoering** [comment]
The source metrics table shows number of broken decisions. Should we add one for sectors too?
https://www.checklistbank.org/catalogue/3/sources

**Comment by yroskov** [comment]
I do not see a usefulness of such reports https://www.checklistbank.org/catalogue/3/sources, at least as they appears through the interface.

**Comment by mdoering** [comment]
The decision counts on the source table are the Toal decisions existing, not the number of broken ones.
I think we need a page for that though for a project, list number of sector, decisions and estimates together with number of broken ones in brackets similar to how we show the changes in last attempts:

<img width="1256" alt="image" src="https://github.com/CatalogueOfLife/data/assets/327505/81f620a1-6099-4653-93ce-79f9fb4e3a56">

I would add that as a different view on the sources project page

**Comment by yroskov** [comment]
Summary table would be helpful. Especially for CoL id3

The problem is, number of sectors even does not shown on GSD sector page:
https://www.checklistbank.org/catalogue/3/sector?offset=0&subjectDatasetKey=1008

**Comment by yroskov** [comment]
_What kind of reports on broken decisions might be helpful?_ 

(1) I need filters for different "classes" of broken decisions, and be able to apply balk decisions to these "classes". 
The most obvious class is a list of "orphan ambiguous synonyms" (cases, where second name disappears in the source, and the remaining name requires disabling the decision on ambiguous synonym). 

Unfortunately, other "classes" among broken decisions are not obvious to me. 

(2) [what else may make reports useful (i.e. transparent and operational for repairing broken decisions?]

**Comment by yroskov** [comment]
>_How do you determine if decisions, sectors or estimates are broken and need attention?_

This button on Tasks page alerts me about **broken decisions**:
![image](https://github.com/CatalogueOfLife/data/assets/28574252/20987466-c39d-49f6-9e84-d3197154b8c2)
https://www.checklistbank.org/catalogue/3/dataset/1130/tasks

However, this list is helpless:
https://www.checklistbank.org/catalogue/3/decision?broken=true&limit=100&offset=0&subjectDatasetKey=1130

Almost all broken decisions are unrepeatable through Rematch and Rematch All buttons. I am just deleting them

I get alert on **broken sectors** through this page https://www.checklistbank.org/catalogue/3/sector?broken=true&offset=0&subjectDatasetKey=1141. 
I evaluate this report when completing a GSD update before syncing. 
(Not all broken sectors need a repair. In my WP example, CoL of December will keep 3 families from the previous sync).

I am not dealing with **estimates**. The pilot project was closed in 2019, and left in the care of Taxonomy Group.

**Comment by mdoering** [comment]
I have created a secret page that shows such a table. I find it pretty useful:
https://www.checklistbank.org/catalogue/3/sources2

@thomasstjerne my suggestion is to integrate all datasource views into a single page where you can then select different views.

**Comment by mdoering** [comment]
<img width="1390" alt="image" src="https://github.com/CatalogueOfLife/data/assets/327505/23be8f88-02f8-44cd-b35c-712a9e47d52e">

**Comment by yroskov** [comment]
>You can see here that there are 4 sources with broken sectors. Not good!

World Plants, Species Fungorum Plus have strong reasons for preserved broken sectors. If I fix them, CoL will loose some taxa, which are not present in current versions. 

IRMNG has own story. We unable to update it due to many disparities between "old" CoL & "new" IRMING.  

I'll have look what's happened to Trichomycetes.

**Comment by yroskov** [comment]
Do you mean sectors lock in Col-3?

![image](https://github.com/CatalogueOfLife/data/assets/28574252/75c007df-7cf8-4839-b7ba-b954ce05f312)

Not sure. This will be an unnecessary complication, plus it will bury the problem (which is visible now through broken sectors). For example, World Plants broken sectors in December will disappear with a new update (i.e. corrupted source will be fixed). As soon as CoL Protozoa & Chromista classification will be revised by TG, it will accommodate latest IRMNG data in matched sectors. Etc.

**Comment by mdoering** [comment]
Ah that makes sense. I thought maybe you just did not want to follow some updates because the data got worse and you then manually "broke" the target so it doesn't sync any longer when you sync an entire source dataset. If that's needed I would suggest to have a lock - which @thomasstjerne could also display in the tree with an icon above, yes
